beetsplug/acousticbrainz.py

```diff
@@ -15,321 +15,28 @@
-        # composites attributes straight away and populating the dictionary
-        # `composites` with the composite attributes.
-
-        # When we are finished traversing `scheme`, `composites` should
-        # map each composite attribute to an ordered list of the values
-        # belonging to the attribute, for example:
-        # `composites = {'initial_key': ['B', 'minor']}`.
-
-        # The recursive traversal.
-        composites = defaultdict(list)
-        yield from self._data_to_scheme_child(data,
-                                              scheme,
-                                              composites)
-
-        # When composites has been populated, yield the composite attributes
-        # by joining their parts.
-        for composite_attr, value_parts in composites.items():
-            yield composite_attr, ' '.join(value_parts)
-
-    def _data_to_scheme_child(self, subdata, subscheme, composites):
-        """The recursive business logic of :meth:`_map_data_to_scheme`:
-        Traverse two structures of nested dictionaries in parallel and `yield`
-        tuples of corresponding leaf nodes.
-
-        If a leaf node belongs to a composite attribute (is a `tuple`),
-        populate `composites` rather than yielding straight away.
-        All the child functions for a single traversal share the same
-        `composites` instance, which is passed along.
-        """
-        for k, v in subscheme.items():
-            if k in subdata:
-                if type(v) == dict:
-                    yield from self._data_to_scheme_child(subdata[k],
-                                                          v,
-                                                          composites)
-                elif type(v) == tuple:
-                    composite_attribute, part_number = v
-                    attribute_parts = composites[composite_attribute]
-                    # Parts are not guaranteed to be inserted in order
-                    while len(attribute_parts) <= part_number:
-                        attribute_parts.append('')
-                    attribute_parts[part_number] = subdata[k]
-                else:
-                    yield v, subdata[k]
-            else:
-                self._log.warning('Acousticbrainz did not provide info '
-                                  'about {}', k)
-                self._log.debug('Data {} could not be mapped to scheme {} '
-                                'because key {} was not found', subdata, v, k)
-
+        deprecated.func = self.deprecation_message
+        return [deprecated]
 
-def _generate_urls(mbid):
-    """Generates AcousticBrainz end point urls for given `mbid`.
-    """
-    for level in LEVELS:
-        yield ACOUSTIC_BASE + mbid + level
+    def deprecation_message(self, lib, opts, args):
+        ui.print_("This plugin is deprecated. AcousticBrainz "
+                  "is shutting down.")
```

Comment:
Maybe we could be a little more elaborate here. Something similar to what you've put in the changelog sounds a little nicer:

This plugin is deprecated and no longer has any function since the AcousicBrainz project has been shut down.